### PR TITLE
Fixes README.md guide for fish shell

### PR DESCRIPTION
**Pull request by bytefluxio**
Make sure you have checked all steps below.

### Prerequisite
* [x] Please consider implementing the feature as a hook script or plugin as a first step.
  * pyenv has some powerful support for plugins and hook scripts. Please refer to [Authoring plugins](https://github.com/pyenv/pyenv/wiki/Authoring-plugins) for details and try to implement it as a plugin if possible.
  **considered, but not applicable**
* [x] Please consider contributing the patch upstream to [rbenv](https://github.com/rbenv/rbenv), since we have borrowed most of the code from that project. (**not applicable**)
  * We occasionally import the changes from rbenv. In general, you can expect changes made in rbenv will be imported to pyenv too, eventually.
  * Generally speaking, we prefer not to make changes in the core in order to keep compatibility with rbenv.
* [x] My PR addresses the following pyenv issue (if any)
  - None - Thought it would be faster to just do the PR instead of mentioning the issue first

### Description
- [x] Here are some details about my PR

Tried to follow the setup instructions using fish, didn't work, so I went through the init script and noticed that with the parameter1 set to the shell in use, it used the fish commands instead of bash. That made it work. So I'm adjusting the README.md to make it easier for other fish users to get up and running with pyenv

### Tests
- [x] My PR adds the following unit tests (if any)
**none**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -276,7 +276,7 @@ easy to fork and contribute any changes back upstream.
 
         And add this to `~/.config/fish/config.fish`:
         ~~~ fish
-        status is-login; and pyenv init --path | source
+        status is-login; and pyenv init --path fish | source
```
Comment (native-api) [inline README.md:279]: This makes no difference -- since this file can only be run with Fish!

**Comment by native-api** [comment]
`pyenv init --path` autodetects the shell that it runs in and produces the appropriate shell commands. So the proposed change looks redundant -- and also exposes an internal interface (the optional `<shell>` argument) which we'd rather not include into official documentation so that we can change it freely.

Did autodetrection fail for you somehow?

**Comment by bytefluxio** [comment]
I guess it did. Let me check some things and get back to you. My guess is omf pyenf might have put functions somewhere that weren't removed when I uninstalled it. So it does make sense to file a bug issue before doing a report I guess. If it's the auto detect that failed. I'll try to find out why and make a new PR for that.

**Comment by bytefluxio** [comment]
**This is just an update post, no action needs to be taken. This PR will be linked to a bug issue I will create later. Not sure if today though**

```
 ~ $ export PATH="/c/Users/rjwes/.pyenv/shims:${PATH}u^C
 ~ $ status is-login; and pyenv init --path | source
- (line 1): Variables cannot be bracketed. In fish, please use "$PATH".
export PATH="/c/Users/rjwes/.pyenv/shims:${PATH}"
                                          ^
from sourcing file -
source: Error while reading file '<stdin>'
 ✘  ~ $ type -a pyenv
pyenv is /c/Users/rjwes/.pyenv/libexec/pyenv
 ~ $
 ```
 
 there was indeed a function defined in .pyenv/libexec/pyenv.init that fish sourced before. Didn't change anything though.
 So pyenv seems to not recognize that im running fish shell. I'll try to reproduce it.

**Comment by native-api** [comment]
Autodetection checks the name of parent process: https://github.com/pyenv/pyenv/blob/e2125b27501888d4163aa526fbbc69c0c7de5082/libexec/pyenv-init#L42-L47

Looks like you're running Fish under a different name.

**Comment by bytefluxio** [comment]
seems like my ps doesn't have an -o option.

Doesn't seem a pyenv bug, but a windows one according to [this slightly unrelated site](https://www.tutorialfor.com/questions-318000.htm)

There seem to be other options than using the `-o` option to extract the information.

@native-api: Would you accept a pr that changes this so that -o isn't needed or would you just wait for microsoft to fix their error? :)



mintty_ENhtqdk0Zb

mintty_I7cq6rwPzf

**Comment by native-api** [comment]
> @native-api: Would you accept a pr that changes this so that -o isn't needed or would you just wait for microsoft to fix their error? :)

From your screenshots, this seems to be a limitation of Cygwin/MinGW rather than Windows.
We don't officially support Windows, see https://github.com/pyenv/pyenv#homebrew-on-macos.

If you can figure out a reliable way to work with `ps` without `-o`, I can accept a PR -- but you're very likely to see breakages elsewhere.

**Pull request closed without merging**